drivers/hud_twenty_twenty_two_to_twenty_twenty_four/app/models/hud_twenty_twenty_two_to_twenty_twenty_four/aggregated_enrollment/move_enrollment_coc.rb

```diff
@@ -0,0 +1,28 @@
+###
+# Copyright 2016 - 2023 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+module HudTwentyTwentyTwoToTwentyTwentyFour::AggregatedEnrollment
+  class MoveEnrollmentCoC
+    include ::HudTwentyTwentyTwoToTwentyTwentyFour::References
+
+    def process(row)
+      row['EnrollmentCoC'] = entry_enrollment_coc(row['HouseholdID'])
+
+      row
+    end
+
+    private def entry_enrollment_coc(household_id)
+      @entry_enrollment_coc ||= {}.tap do |h|
+        reference(:enrollment_coc) do |row|
+          next unless row['DataCollectionStage'].to_i == 1
+
+          h[row['HouseholdID']] ||= row['CoCCode']
```

Comment:
I think we'll need to handle empty HouseholdID, if we don't, everyone with a nil HouseholdID will end up with the same CoCCode.  Not sure if we should use EnrollmentID as a backup, or just return a blank CoCCode.  Open to suggestions.

Reply:
Good point. This seems like a recipe for making bad DQ even worse...
The 2022 spec says:

> There must be one and only one record in EnrollmentCoC.csv with a DataCollectionStage of 1 for each EnrollmentID in Enrollment.csv where RelationshipToHoH is equal to 1. Additionally, there must be one and only one record in EnrollmentCoC.csv with a DataCollectionStage of 1 for each HouseholdID in Enrollment.csv.

So, I don't think using the EnrollmentID will help much. and I don't think leaving it blank would be making the DQ any worse than it already is.

Alternatively, the 2024 spec for the CoCCode column says:

> Must match a record in ProjectCoC.csv with the same ProjectID

So, we could use the code from ProjectCoC when there is only a single CoC Code for the Project given that_most_ projects have only one, if you think it's worth it.